pydatastructs/strings/suffix_tree.py

```diff
@@ -0,0 +1,144 @@
+from pydatastructs.utils.misc_util import (
+    SuffixTreeNode, SuffixTreeEdge, Suffix, Backend, raise_if_backend_is_not_python)
+
+__all__ = [
+    'SuffixTree'
+]
+
+
+# Ukkonen's algorithm gives a O(n) + O(k) contruction time for a suffix tree,
+# where n is the length of the string and k is the size of the alphabet of that string.
+# Ukkonen's is an online algorithm,
+# processing the input sequentially and producing a valid suffix tree at each character.
+
+
+class SuffixTree(object):
+    """A suffix tree for string matching. Uses Ukkonen's algorithm
```

Comment:
Add docstring for class as well. See an example below,

https://github.com/codezonediitj/pydatastructs/blob/2482adb276055403485266d6d02a1489451c8e76/pydatastructs/trees/binary_trees.py#L24-L53

Also how the names `key`, `root_data` are exactly the same as the parameters of `__new__`,

https://github.com/codezonediitj/pydatastructs/blob/2482adb276055403485266d6d02a1489451c8e76/pydatastructs/trees/binary_trees.py#L59-L60